frontend/pages/ManageControlsPage/Scripts/ScriptBatchDetailsPage/components/ScriptBatchHostsTable/ScriptBatchHostsTable.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useCallback } from "react";
+import { useQuery } from "react-query";
+import { InjectedRouter } from "react-router";
+import { AxiosError } from "axios";
+
+import PATHS from "router/paths";
+
+import scriptsAPI, {
+  IScriptBatchHostResultsResponse,
+  IScriptBatchHostResultsQueryKey,
+  ScriptBatchHostsOrderKey,
+} from "services/entities/scripts";
+import { OrderDirection } from "services/entities/common";
+
+import {
+  SCRIPT_BATCH_HOST_EXECUTED_STATUSES,
+  ScriptBatchHostStatus,
+} from "interfaces/script";
+
+import { DEFAULT_USE_QUERY_OPTIONS } from "utilities/constants";
+import { getNextLocationPath } from "utilities/helpers";
+
+import TableContainer from "components/TableContainer";
+import DataError from "components/DataError";
+import { ITableQueryData } from "components/TableContainer/TableContainer";
+
+import generateColumnConfigs from "./ScriptBatchHostsTableConfig";
+
+export const DEFAULT_SORT_DIRECTION = "asc";
+export const DEFAULT_PAGE_SIZE = 20;
+export const DEFAULT_SORT_COLUMN = "display_name";
+
+const baseClass = "script-batch-hosts-table";
+
+interface IScriptBatchHostsTableProps {
+  batchExecutionId: string;
+  selectedHostStatus: ScriptBatchHostStatus;
+  page: number;
+  orderDirection: OrderDirection;
+  orderKey: ScriptBatchHostsOrderKey;
+  setHostScriptExecutionIdForModal: (id: string) => void;
+  router: InjectedRouter;
+}
+
+const ScriptBatchHostsTable = ({
+  batchExecutionId,
+  selectedHostStatus,
+  page,
+  orderDirection,
+  orderKey,
+  setHostScriptExecutionIdForModal,
+  router,
+}: IScriptBatchHostsTableProps) => {
+  const perPage = DEFAULT_PAGE_SIZE;
+  const { data: hostResults, isLoading, error } = useQuery<
+    IScriptBatchHostResultsResponse,
+    AxiosError,
+    IScriptBatchHostResultsResponse,
+    IScriptBatchHostResultsQueryKey[]
+  >(
+    [
+      {
+        scope: "script_batch_host_results",
+        batch_execution_id: batchExecutionId,
+        status: selectedHostStatus,
+        page,
+        per_page: perPage,
+        order_direction: orderDirection,
+        order_key: orderKey,
+      },
+    ],
+    ({ queryKey }) => scriptsAPI.getScriptBatchHostResults(queryKey[0]),
+    {
+      ...DEFAULT_USE_QUERY_OPTIONS,
+    }
+  );
+
+  const handleRowClick = useCallback(
+    (row: any) => {
+      if (SCRIPT_BATCH_HOST_EXECUTED_STATUSES.includes(selectedHostStatus)) {
+        setHostScriptExecutionIdForModal(row.original.script_execution_id);
+      } else {
+        router.push(PATHS.HOST_DETAILS(row.original.id));
+      }
+    },
+    [router, selectedHostStatus, setHostScriptExecutionIdForModal]
+  );
```

Comment:
I see now why you were asking about this. I don't see anything in the Figma about clicking on the row -- I only see notes about clicking on the host name and on the "script details" button that appears on hover for certain statuses.  Is this something you discussed separately w/ product?

Reply:
Yup, missed that note in the Figma for sure 👍